openadapt/strategies/mixins/svg.py

```diff
@@ -18,9 +20,31 @@ class SVGReplayStrategyMixin(BaseReplayStrategy):
 
     Attributes:
         recording: the recording to be played back
-
+        path_to_vtracer: the absolute path to the cargo command
     """
-    def __init__(self, recording: Recording):
+    def __init__(self, recording: Recording, vtracer_executable: str = "vtracer"):
+        if not is_dependency_installed(vtracer_executable):
+            if not is_dependency_installed("cargo"):
+                root_directory = os.path.expanduser('~')
+                
+                # install rust
+                if platform.system() == "Darwin":
+                    subprocess.run(["brew", "install", "rust"])
+                elif platform.system() == "Windows":
+                    # windows
+                    cwd = os.getcwd()
+                    os.chdir(root_directory)
+                    subprocess.run(["curl", "--proto", "'=https'", "--tlsv1.2", "-sSf", "https://sh.rustup.rs", "|", "sh"])
+                    os.chdir(cwd)
```

Comment:
Thank you @dianzrong !

Can you please refactor this so that we can write:

```
if not is_executable("cargo"):
    install_executable("cargo")
```
`dependencies.py`:
```
DEP_NAME_TO_SYS_TO_INSTALL_CMD = {
    "cargo": {
        "darwin": {
            "brew install rust",
        },
        "windows": {
            "curl --proto ...",
        }
     },
     ...
}
def install_executable("cargo"):
    ...
```

Reply:
Addressed here [28f661f](https://github.com/OpenAdaptAI/OpenAdapt/pull/198/commits/28f661fc9403c52df51f476c314bd45e6e9faba9)